src/main/java/org/terracotta/statistics/derived/latency/LatencyPeriodAccumulator.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * All content copyright Terracotta, Inc., unless otherwise indicated.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terracotta.statistics.derived.latency;
+
+import org.terracotta.statistics.observer.ChainedEventObserver;
+
+/**
+ * @author Mathieu Carbou
+ */
+public class LatencyPeriodAccumulator implements LatencyStatistic, ChainedEventObserver {
+
+  private final LatencyAccumulator accumulator;
+  private final long start;
+  private final long end;
+
+  LatencyPeriodAccumulator(long start, long length, long... latencies) {
+    this.start = start;
+    this.end = start + length;
+    accumulator = LatencyAccumulator.accumulator(latencies);
+  }
+
+  public boolean isBefore(long time) {
```

Comment:
It's more an isEqualsOrBefore

Reply:
no because end is exclusive: `[start;end[`. The time matching `end` would be included in the next accumulator which starts at `end`.

Reply:
I know. It's just the method name that I don't like because it accepts equal values, not only before values.

Reply:
I don't understand: this tests if a range is before a specific time. This is the equivalent of Joda's time interval.isBefore(time) 
http://joda-time.sourceforge.net/apidocs/org/joda/time/ReadableInterval.html#isBefore(org.joda.time.ReadableInstant)

Reply:
Hum... interesting... Let's follow Joda time